databricks/koalas/groupby.py

```diff
@@ -2155,6 +2155,76 @@ def expanding(self, min_periods=1):
         """
         return ExpandingGroupby(self, min_periods=min_periods)
 
+    def get_group(self, name):
+        """
+        Construct DataFrame from group with provided name.
+
+        Parameters
+        ----------
+        name : object
+            The name of the group to get as a DataFrame.
+
+        Returns
+        -------
+        group : same type as obj
+
+        Examples
+        --------
+        >>> kdf = ks.DataFrame([('falcon', 'bird', 389.0),
+        ...                     ('parrot', 'bird', 24.0),
+        ...                     ('lion', 'mammal', 80.5),
+        ...                     ('monkey', 'mammal', np.nan)],
+        ...                    columns=['name', 'class', 'max_speed'],
+        ...                    index=[0, 2, 3, 1])
+        >>> kdf
+             name   class  max_speed
+        0  falcon    bird      389.0
+        2  parrot    bird       24.0
+        3    lion  mammal       80.5
+        1  monkey  mammal        NaN
+
+        >>> kdf.groupby("class").get_group("bird").sort_index()
+             name class  max_speed
+        0  falcon  bird      389.0
+        2  parrot  bird       24.0
+
+        >>> kdf.groupby("class").get_group("mammal").sort_index()
+             name   class  max_speed
+        1  monkey  mammal        NaN
+        3    lion  mammal       80.5
+        """
+        groupkeys = self._groupkeys
+        if not is_hashable(name):
+            raise TypeError("unhashable type: '{}'".format(type(name).__name__))
+        elif len(groupkeys) > 1 and not isinstance(name, tuple):
+            raise ValueError("must supply a tuple to get_group with multiple grouping keys")
+        if isinstance(name, str):
+            name = [name]
+        cond = F.lit(True)
+        for groupkey, item in zip(groupkeys, name):
+            scol = groupkey.spark.column
+            cond = cond & (scol == item)
+        internal = self._kdf._internal.with_filter(cond)
+        if internal.spark_frame.head() is None:
+            raise KeyError(name)
+        if self._agg_columns_selected:
+            spark_frame = internal.spark_frame.select(
+                internal.index_spark_columns + self._agg_columns_scols
+            )
+            column_labels = list()
+            for agg_column in self._agg_columns:
+                agg_column_name = agg_column.name
+                if isinstance(agg_column_name, tuple):
+                    column_labels.append(agg_column_name)
+                else:
+                    column_labels.append((agg_column_name,))
+
+            internal = InternalFrame(
+                spark_frame=spark_frame, index_map=internal.index_map, column_labels=column_labels,
+            )
```

Comment:
Can we drop column names? Could you confirm?

Reply:
Maybe do you mean this ?

```python
>>> kdf
Hello        A       B         C
Koalas    name   class max_speed
0       falcon    bird     389.0
2       parrot    bird      24.0
3         lion  mammal      80.5
1       monkey  mammal       NaN

>>> result = kdf.groupby(("B", "class"))[[("A", "name")]].get_group("mammal")
>>> result
        A
     name
3    lion
1  monkey

>>> result._internal.column_labels
[('A', 'name')]

>>> result._internal.column_label_names
[None, None]
```

Reply:
yep, is it fine to remove the names?

Reply:
Oh, we should preserve it. Thanks for figuring out .. !